Client/Frontend/Home/HistoryPanel.swift

```diff
@@ -122,6 +122,8 @@ class HistoryPanel: SiteTableViewController, HomePanel {
     }
 
     func notificationReceived(_ notification: Notification) {
+        reloadData()
```

Comment:
imo. I think this should be wrapped in a main thread rather than the caller of the notification. For example, if someone else posts `NotificationPrivateDataClearedHistory` they might forget to call it on the main thread. What do you think?

Reply:
A func calling other Client code (i.e another func non-private to caller) off-main upon completion of work I think  is a highly exceptional case or a coding error. Client is written entirely as main-thread code: no mutex'd memory access, no thread guards on on UIKit calls. 
I would hope we would be super-strict about enforcing containment of off-main thread calls to the minimum subset of code necessary.

Reply:
Yea. I think we try our best to be super strict about this. And XCode usually catches us on this as well.